src/gt4py/next/program_processors/runners/dace_iterator/__init__.py

```diff
@@ -111,12 +111,35 @@ def get_stride_args(
     return stride_args
 
 
-_build_cache_cpu: dict[int, CompiledSDFG] = {}
-_build_cache_gpu: dict[int, CompiledSDFG] = {}
-
-
-def get_cache_id(*cache_args) -> int:
-    return sum([hash(str(arg)) for arg in cache_args])
+_build_cache_cpu: dict[str, CompiledSDFG] = {}
+_build_cache_gpu: dict[str, CompiledSDFG] = {}
+
+
+def get_cache_id(
+    program: itir.FencilDefinition,
+    arg_types: Sequence[ts.TypeSpec],
+    column_axis: Optional[Dimension],
+    offset_provider: Mapping[str, Any],
+) -> str:
+    max_neighbors = [
+        (k, v.max_neighbors)
+        if isinstance(v, (NeighborTableOffsetProvider, StridedNeighborOffsetProvider))
+        else (k, -1)
+        for k, v in offset_provider.items()
+    ]
```

Comment:
Just drop the ones that don't have any compile-time properties. The ones with only runtime properties should manifest in the SDFG structure, that should be enough, right? Let me know if I'm missing something here.
```suggestion
    max_neighbors = [
        (k, v.max_neighbors)
        for k, v in offset_provider.items()
        if isinstance(v, (NeighborTableOffsetProvider, StridedNeighborOffsetProvider))
    ]
```